source/common/router/config_impl.cc

```diff
@@ -643,15 +706,56 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
     }
   }
 
+  if (path_rewrite_policy_.enabled()) {
+    if (!prefix_rewrite_.empty() || route.route().has_regex_rewrite()) {
+      throw EnvoyException(
+          "Specify only one of prefix_rewrite, regex_rewrite or path_rewrite_policy");
+    }
+  }
+
+  if (!prefix_rewrite_.empty()) {
+    if (route.route().has_regex_rewrite() || path_rewrite_policy_.enabled()) {
+      throw EnvoyException(
+          "Specify only one of prefix_rewrite, regex_rewrite or path_rewrite_policy");
+    }
+  }
+
   if (route.route().has_regex_rewrite()) {
-    if (!prefix_rewrite_.empty()) {
-      throw EnvoyException("Cannot specify both prefix_rewrite and regex_rewrite");
+    if (!prefix_rewrite_.empty() || path_rewrite_policy_.enabled()) {
+      throw EnvoyException(
+          "Specify only one of prefix_rewrite, regex_rewrite or path_rewrite_policy");
     }
+  }
+
+  if (route.route().has_regex_rewrite()) {
     auto rewrite_spec = route.route().regex_rewrite();
     regex_rewrite_ = Regex::Utility::parseRegex(rewrite_spec.pattern());
     regex_rewrite_substitution_ = rewrite_spec.substitution();
   }
 
+  // Check if pattern rewrite is enabled that the pattern matching is also enabled.
+  // This is needed to match up variable values.
+  if (path_rewrite_policy_.enabled() &&
+      path_rewrite_policy_.predicate()->name() == Extensions::PatternTemplate::Rewrite::NAME) {
+    if (!path_match_policy_.enabled() ||
+        path_match_policy_.predicate()->name() != Extensions::PatternTemplate::Match::NAME) {
+      throw EnvoyException(fmt::format("unable to use {} extension without {} extension",
+                                       Extensions::PatternTemplate::Rewrite::NAME,
+                                       Extensions::PatternTemplate::Match::NAME));
+    }
+  }
+
+  // Validation between extensions as they share rewrite pattern variables.
+  if (path_rewrite_policy_.enabled() && path_match_policy_.enabled()) {
+    if (!Extensions::PatternTemplate::isValidSharedVariableSet(
```

Comment:
Related to the previous comment, I see that this rewrite predicate needs to validate that capture groups were correctly defined in the match extension, but as written we'd have to update this for any new pairs of match and rewrite extensions.

Reply:
We can add an if to only check in case of these two extension types.